### PR TITLE
Implement SessionBlocks and make rooms outside their block stale

### PR DESCRIPTION
**Pull request by oliverlevay**

### CODE REVIEW

File: src/main/scala/model.scala [hunk 1]
```diff
@@ -87,7 +92,18 @@ case class Room(
 
   def popApprovalQueue(): Room = copy(approvalQueue = approvalQueue.drop(1))
 
-  def isExpired: Boolean = created < Date.now.minusHours(Room.HoursUntilExpired)
+  def isExpired(): Boolean = {
+    val expiredByMaximumTime = created < Date.now.minusHours(Room.HoursUntilExpired)
+    val sessionBlock = SessionBlock.sessionBlocks.find((block) => {
+      block.start >= created.hour && block.end <= created.hour
+    })
+    if(sessionBlock.isEmpty) {
```
Comment (bjornregnell) [inline src/main/scala/model.scala:100]: Minor itch: I want spaces after if

Comment (oliverlevay) [reply src/main/scala/model.scala]: Fixat!

**Comment by bjornregnell** [comment]
Thanks! I remember vaguely that there was issues with cancel directly after a session (I had 3 hours-cancel and then nightly wipes). Lets encourage some experienced supervisors to chip in and review this. I'll post in our Discord.

**Comment by bjornregnell** [comment]
I will close this as I have another solution coming.... We will allow many supervisors in each room. I also think it is too brutal to auto-clear a room that might still have people lagging.

**Pull request closed without merging**